secure-api-gateway-ig-extensions/src/main/java/com/forgerock/sapi/gateway/dcr/sigvalidation/RegistrationRequestJwtSignatureValidationFilter.java

```diff
@@ -0,0 +1,226 @@
+                registrationRequestJwtValidator.validateJwtSignature(fapiInteractionId, registrationRequest).thenAsync(
+                    regRequestValidationResponse -> {
+                        if (regRequestValidationResponse.getStatus() != Status.OK) {
+                            log.info("({}) Response from validation is not OK as expected. Resposne: {}",
+                                    fapiInteractionId, regRequestValidationResponse);
+                            return Promises.newResultPromise(response);
+                        }
+                        log.debug("({}) Registration Request and embedded SSA both have valid signatures",
+                                fapiInteractionId);
+                        return next.handle(context, request);
+                    }, ex -> {
+                        log.info("({}) Registration Request validation failed: {}", fapiInteractionId,
+                                ex.getMessage(), ex);
+                        Response badRequest = new Response(Status.BAD_REQUEST).setEntity(getJsonResponseBody(ex));
+                        return Promises.newResultPromise(badRequest);
+                    }, rte -> {
+                        log.info("({}) A Runtime Exception occurred while validating the Registration Response Jwt " +
+                                "signature: {}", fapiInteractionId, "error: " + rte.getMessage(), rte);
+                        return Promises.newResultPromise(new Response(Status.INTERNAL_SERVER_ERROR));
+                    }
+                ), ex -> {
+                    log.info("({}) Software Statement validation failed:  {}", fapiInteractionId, ex.getMessage(),
+                            ex);
+                    Response badRequest = new Response(Status.BAD_REQUEST).setEntity(getJsonResponseBody(ex));
+                    return Promises.newResultPromise(badRequest);
+                }, rte -> {
+                    log.error("({}) Runtime Error while validating Registration Request: {}", fapiInteractionId,
+                            "error: " + rte.getMessage(), rte);
+                    return Promises.newResultPromise(new Response(Status.INTERNAL_SERVER_ERROR));
+                });
+        } catch (RuntimeException rte){
+            log.error("({}) Runtime Error occurred in RequestAndSsaSignatureValidationFilter: {}", fapiInteractionId,
+                    rte.getMessage(), rte);
+            return Promises.newResultPromise(new Response(Status.INTERNAL_SERVER_ERROR));
+        }
+    }
+
+    private String getJsonResponseBody(DCRSignatureValidationException ex) {
+        return "{\"error_code\":\"" + ex.getErrorCode().getCode() + "\"," +
+                "\"error_description\":\"" + ex.getErrorDescription() + "\"}";
+    }
+
+
+    /**
+     * Heaplet is used to get arguments from the IG config and create a RequestAndSsaSignatureValidationFilter
+     * on the IG heap.
+     */
+    public static class Heaplet extends GenericHeaplet {
```

Comment:
Document how to create the filter via config, see com.forgerock.sapi.gateway.mtls.TransportCertValidationFilter.Heaplet for an example